app.py

```
@@ -22,76 +26,192 @@ def initialize_session_state():
         try:
-            if not firecrawl_api_key:
-                st.sidebar.error("❌ Firecrawl API Key is required")
+            if not st.session_state.firecrawl_api_key:
+                container.error("❌ Firecrawl API Key is required")
             else:
-                st.session_state.browser = BrowserAutomation(api_key=firecrawl_api_key)
+                st.session_state.browser = BrowserAutomation(api_key=st.session_state.firecrawl_api_key)
                 st.session_state.browser.start_browser()
-                st.sidebar.success("✅ Browser started")
+                container.success("✅ Browser started")
         except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)